scenarioo-client/app/shared/navigation/branchesAndBuilds.service.js

```diff
@@ -57,7 +44,118 @@ angular.module('scenarioo.services').factory('BranchesAndBuildsService', functio
         return deferred.promise;
     };
 
+    function getBranch(loadedData, branchName) {
+        var index;
+        for (index = 0; index < loadedData.branches.length; index++) {
+            if (loadedData.branches[index].branch.name === branchName) {
+                return loadedData.branches[index];
+            }
+        }
+
+    }
+
+    function getBuild(branch, buildName) {
+        var index;
+        if (angular.isDefined(branch)) {
+            var allBuildsOnSelectedBranch = branch.builds;
+            for (index = 0; index < branch.builds.length; index++) {
+                if (allBuildsOnSelectedBranch[index].linkName === buildName) {
+                    return allBuildsOnSelectedBranch[index];
+                }
+            }
+        }
+    }
+
+    function getBranchDisplayName(wrappedBranch) {
+
+        if (wrappedBranch === undefined) {
+            return null;
+        }
+
+        var displayName = wrappedBranch.branch.name;
+        if (wrappedBranch.alias) {
+            displayName = displayName + ' (' + wrappedBranch.branch.description + ')';
+        }
+        return displayName;
+    }
+
+    function getDisplayNameForBuildName(branchName, buildName, returnShortText) {
+        var deferred = $q.defer();
+
+
+        getBranchesAndBuildsData().then(function onSuccess(result){
```

Comment:
We probably don't. Also here just an existing pattern was used. If we want to cache it in the service, we have to think when to invalidate it. Not only for diff view features but also for the existing code.
I also agree that flag parameters are a bad choice but then again, that's existing code. They did not change it. I'm not sure if we should also refactor existing code in this PR.